plugins/flashrom/fu-flashrom-device.c

```diff
@@ -48,11 +72,206 @@ fu_flashrom_device_set_quirk_kv (FuDevice *device,
 	return FALSE;
 }
 
+static int
+fu_flashrom_device_debug_cb (enum flashrom_log_level lvl, const char *fmt, va_list args)
+{
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wformat-nonliteral"
+	g_autofree gchar *tmp = g_strdup_vprintf (fmt, args);
+#pragma clang diagnostic pop
+	g_autofree gchar *str = fu_common_strstrip (tmp);
+	if (g_strcmp0 (str, "OK.") == 0 || g_strcmp0 (str, ".") == 0)
+		return 0;
+	switch (lvl) {
+	case FLASHROM_MSG_ERROR:
+	case FLASHROM_MSG_WARN:
+		g_warning ("%s", str);
+		break;
+	case FLASHROM_MSG_INFO:
+		g_debug ("%s", str);
+		break;
+	case FLASHROM_MSG_DEBUG:
+	case FLASHROM_MSG_DEBUG2:
+		if (g_getenv ("FWUPD_FLASHROM_VERBOSE") != NULL)
+			g_debug ("%s", str);
+		break;
+	case FLASHROM_MSG_SPEW:
+		break;
+	default:
+		break;
+	}
+	return 0;
+}
+
+static gboolean
+fu_flashrom_device_setup (FuDevice *device, GError **error)
+{
+	FuFlashromDevicePrivate *priv = GET_PRIVATE (FU_FLASHROM_DEVICE (device));
+	gint rc;
+
+	if (flashrom_init (SELFCHECK_TRUE)) {
```

Comment:
won't it be problematic to call `flashrom_init` multiple times?  I would think you want to once at plugin init, but not at a device level.

Reply:
Agree.

Reply:
@sammc @bleungatchromium do you have any insights on this?

I tested this with more than one device (i.e., calling init at least twice) and everything worked as expected buy I'm not sure if it is the right thing to do.

Reply:
PTAL. I moved it to fu_plugin_startup() so we can fail (init is void)